src/tools/rados/rados.cc

```diff
@@ -2116,6 +2116,7 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
 
   if (create_pool && !pool_name) {
     cerr << "--create-pool requested but pool_name was not specified!" << std::endl;
+    rados.shutdown();
```

Comment:
i think an alternative approach is to instead call usage(cerr) and return `1`, and leave the cleanup to the destructor of `rados`.

Reply:
i also considered this, and thought either way is ok. If you insist, i could revise it